odoo/models.py

```diff
@@ -5657,28 +5657,35 @@ def _cache(self):
         """ Return the cache of ``self``, mapping field names to values. """
         return RecordCache(self)
 
-    @api.model
     def _in_cache_without(self, field, limit=PREFETCH_MAX):
         """ Return records to prefetch that have no value in cache for ``field``
             (:class:`Field` instance), including ``self``.
             Return at most ``limit`` records.
         """
+        recs = self.browse(unique(self._prefetch_ids))
+        ids = self.env.cache.get_missing_ids(recs, field)
```

Comment:
Why not `self.env.cache.get_missing_ids(recs - self, field)` anymore ?

Reply:
because the deduplication in `_batch_with` is more efficient than a recordset difference